src/BackwardCompatibility/SalesChannelServiceCompilerPass.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Omikron\FactFinder\Shopware6\BackwardCompatibility;
+
+use Omikron\FactFinder\Shopware6\Export\SalesChannelService;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+class SalesChannelServiceCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if ($container->has('Shopware\Core\System\SalesChannel\Context\CachedSalesChannelContextFactory')) {
```

Comment:
This will work only on 6.4 but what about earlier versions? 
There has to be some `else` added here defining the service for earlier version

Reply:
I'm not sure that we need `else` condition. As far as I remeber, `CachedSalesChannelContextFactory` was introduced in SW 6.4, so on any other version `if` condition will return `false` and `SalesChannelService` will be initiate with SalesChannelContextFactory parameter.

Reply:
that's true but right now `SalesChannelService` we expect to get `SalesChannelContextFactoryInterface` implementation which core `SalesChannelContextFactory` does not implement, hence no autowiring.
![image](https://user-images.githubusercontent.com/4511883/118782401-596b4080-b88e-11eb-8f1a-025ad4328918.png)